sdk/storage/Azure.Storage.Files.Shares/src/Models/ShareFileUploadOptions.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+
+namespace Azure.Storage.Files.Shares.Models
+{
+    /// <summary>
+    /// Options for uploading a file.
+    /// </summary>
+    public class ShareFileUploadOptions
+    {
+        /// <summary>
+        /// Constructs options instance using a stream as content source.
+        /// </summary>
+        /// <param name="stream">Content stream.</param>
+        public ShareFileUploadOptions(Stream stream)
```

Comment:
Why is `stream` in the constructor instead of being settable?

Reply:
I recall seeing a pattern in our libraries where required parameters are in options constructors instead of the method signature. e.g. the above instead of `Upload(Stream stream, ShareFileUploadOptions options)`. Which tends to be nice when introducing different required parameter sets; adding a `string path` alternative to providing a stream is just another constructor on the class instead of two new overloads on the client.